ui-v2/tests/unit/mixins/acl/with-actions-test.js

```diff
@@ -18,3 +28,64 @@ test('it works', function(assert) {
   const subject = this.subject();
   assert.ok(subject);
 });
+test('use persists the token and calls transitionTo correctly', function(assert) {
+  assert.expect(4);
+  this.register(
+    'service:feedback',
+    Service.extend({
+      execute: function(cb, name) {
+        assert.equal(name, 'use');
+        return cb();
+      },
```

Comment:
Why not use sinon here instead of putting the assertions in the methods?

Reply:
Of the top of my head I’m not massively clear how I would do this using sinon, could you show me what you were thinking? (I'm gonna merge this BTW, but happy to keep chatting about this, can still put another PR in ontop)

Reply:
The simplest way would be something like this:

```js
  const executeSpy = sinon.spy();
  this.register(
    'service:feedback',
    Service.extend({
      execute: function(cb, name) {
        executeSpy(name);
        return cb();
      },
    })
  );
```

But there may be a more elegant way using something like `sinon.spy(NewService, 'execute')`. I'm not as familiar with that API for spies and it could very well not work with Ember Objects.

Reply:
Oh cool, ok sorry I thought you meant replacing the entire `execute` function with a single spy. I can drop that ^ in if you think its nicer?

Reply:
Eh, _I_ think it's nicer, but it's not exactly convention. I just find counting the expected number of assertions and having said assertions have mismatched line and execution order to be kinda hacky.

It's your call to make.

Reply:
Cool, I'm easy, will drop it in then, probably tomorrow now. Will ping you on the PR

Reply:
Hey @DingoEatingFuzz 

I had another look at this this morning and a think about what you said.

I noticed that the final assertions are done in a callback (a promise `then`):

https://github.com/hashicorp/consul/blob/7a925b4188230d3f3c99c56fac60e9b93785a6dc/ui-v2/tests/unit/mixins/acl/with-actions-test.js#L55-L60

So I'd still want to be checking the number of expected assertions, whether I did it with spies or just assertions. Plus if you think that it's not exactly convention, maybe I should leave as is? As it is currently achieves the same thing but with less 'stuff', and I think with unit tests its always best to try to stay as 'close to the bone' and use as little 'bells and whistles' as possible, if I can achieve the same thing without using `sinon` then why not?

Cheers,

John